pkg/apiclient/apiclient.go

```diff
@@ -64,9 +64,6 @@ func NewClient(argoServer string, authSupplier func() string, clientConfig clien
 
 func NewClientFromOpts(opts Opts) (context.Context, Client, error) {
 	log.WithField("opts", opts).Debug("Client options")
-	if opts.Offline {
-		return newOfflineClient(opts.OfflineFiles)
-	}
```

Comment:
You can't just ignore `--offline` option, it will fail back to `newArgoKubeClient` and won't work without configuring the kubernetes.
Here's the error I've got running the command with your change, without any k8s/k8s-alternative configured.
```sh
$ ./dist/argo lint ./examples/*.yaml --offline
Error: invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variabl
```